src/inference/graph_inference.cpp

```diff
@@ -247,7 +241,7 @@ class GraphQuery : public Nice2Query {
   };
 
   std::vector<std::vector<Arc> > arcs_adjacent_to_node_;
-  std::vector<std::vector<Factor> > factors_of_a_node_;
+  std::vector<std::vector<Factor*> > factors_of_a_node_;
```

Comment:
Instead of Factor*, you can use int to point to the index in the factors_ vector. This avoids possible bugs (in the future) in case somebody decides to add a factor and the vector is resized (the pointers will be invalidated then).